frontend/src/routes/Salary.jsx

```diff
@@ -1,17 +1,29 @@
-import { Select, Button, Typography } from "antd";
+import { Select, Button, Typography, Table } from "antd";
 import { SearchOutlined } from "@ant-design/icons";
 import axios from "axios";
 import { useEffect, useState } from "react";
-import { Table } from "antd";
 import renderFormattedNumber from "../helpers";
+import CreateCsv from "../components/CreateCsv";
 
 const { Title } = Typography;
 
 const Salary = () => {
   const [salaryDates, setSalaryDates] = useState([]);
   const [selectedSalaryDate, setSelectedSalaryDate] = useState(null);
-  const [columnNames, setColumnNames] = useState([]);
+  const [columns, setColumns] = useState([]);
   const [tableData, setTableData] = useState([]);
+
+  let finnishDate = "";
+  let finnishDateNoDots = "";
+
+  if (selectedSalaryDate) {
+    let year = selectedSalaryDate.slice(0, 4);
+    const month = selectedSalaryDate.slice(5, 7);
+    const day = selectedSalaryDate.slice(8, 10);
+    finnishDate = day + "." + month + "." + year;
+    finnishDateNoDots = year + month + day;
```

Comment:
Just fyi: another way would be to parse the string as a date using `new Date(selectedSalaryDate)`, then you could use date methods to do the conversion, however plain javascript dates don't have nice methods for this (you would need a library such as moment to do formatting easily). So this solution is good.